hardware/stm32f446xx/board_def.h

```diff
@@ -42,12 +42,22 @@
 // Number of ADC channels
 #define ADC_NUM_CHANNELS 16
 
-// ADC resolution in bits
-#define ADC_RESOLUTION 12
 // Maximum ADC value
 #define ADC_MAX_VALUE ((1 << ADC_RESOLUTION) - 1)
 
 #if !defined(ADC_NUM_SAMPLE_CYCLES)
 // Number of sample cycles for each ADC conversion
 #define ADC_NUM_SAMPLE_CYCLES ADC_SAMPLETIME_3CYCLES
 #endif
+
+// ADC resolution in bits, set in the mcu driver and keyboard json
+#if !defined(ADC_RESOLUTION)
+#define ADC_RESOLUTION MAX_ADC_RESOLUTION
+#define ADC_RESOLUTION_HAL ADC_RESOLUTION_12B
+#elif ADC_RESOLUTION == 8
+#define ADC_RESOLUTION_HAL ADC_RESOLUTION_8B
+#elif ADC_RESOLUTION == 10
+#define ADC_RESOLUTION_HAL ADC_RESOLUTION_10B
+#elif ADC_RESOLUTION == 12
+#define ADC_RESOLUTION_HAL ADC_RESOLUTION_12B
+#endif
```

Comment:
I think we should add an else statement here to raise compile time error for catching the user from using unsupported ADC resolution.